plugins/keyboard-navigation/src/navigation.js

```diff
@@ -0,0 +1,377 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Holds all methods necessary to use Blockly through the
+ * keyboard.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+import * as Blockly from 'blockly/core';
+import * as Constants from './constants';
+import {
+  registrationName as cursorRegistrationName,
+  registrationType as cursorRegistrationType} from './flyout_cursor';
+
+/**
+ * Class that holds all methods necessary for keyboard navigation to work.
+ */
+export class Navigation {
+  /**
+   * Constructor for keyboard navigation.
+   */
+  constructor() {
+    /**
+     * Object holding the location of the cursor for each workspace.
+     * Possible locations of the cursor are: workspace, flyout or toolbox.
+     * @type {Object<string,Constants.STATE>}
+     * @protected
+     */
+    this.workspaceStates = {};
+
+    /**
+     * An optional method that allows a developer to customize how to handle
+     * logs, warnings, and errors. The first argument is one of 'log', 'warn',
+     * or 'error'. The second argument is the message.
+     * @type {?function(string, string)}
+     * @public
+     */
+    this.loggingCallback = null;
+
+    /**
+     * The distance to move the cursor when the cursor is on the workspace.
```

Comment:
Ok, I added. It is a bit weird since they should be in workspace coordinates I think since we are positioning something that is on the workspace and not above it. However, we don't know the workspace  scale here so they have to be converted to workspace coordinates when they are used.